src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -316,6 +361,19 @@ bool AutoDownloadRule::matches(const QString &articleTitle) const
         return false;
     }
 
+    if (m_dataPtr->smartFilter) {
```

Comment:
use `useSmartFilter()` instead of accessing `m_dataPtr` directly. Same everywhere else that is accessing values directly through `m_dataPtr` instead of the getter.